### PR TITLE
Bench: Add new device endpoint to create new device with defaults (#195)

### PR DESCRIPTION
**Pull request by ao-david**
an endpoint has been added to /set/devices/new that reads the form fields for
- name
- mac
- wifi
- location
- device type and creates a new device based on these settings. At the moment the only supported device type is a controller.

### CODE REVIEW

File: cmd/oceanbench/set.go [hunk 1]
```diff
@@ -374,6 +376,99 @@ func editDevicesHandler(w http.ResponseWriter, r *http.Request) {
...
+	pass := r.FormValue("pass")
+	sLat := r.FormValue("lat")
+	sLong := r.FormValue("long")
+	r.ParseForm()
+
+	var lat, long float64
+	if sLat != "" && sLong != "" {
+		lat, err = strconv.ParseFloat(sLat, 64)
+		if err != nil {
+			writeError(w, fmt.Errorf("unable to parse float64 from: %s, err:", err))
```
Comment (trekhopton) [inline cmd/oceanbench/set.go:417]: Looks like you were intending to include the variable in the log message but instead you're including the error in the wrong spot.

File: cmd/oceanbench/set.go [hunk 2]
```diff
@@ -374,6 +376,99 @@ func editDevicesHandler(w http.ResponseWriter, r *http.Request) {
...
+			return
+		}
+		long, err = strconv.ParseFloat(sLong, 64)
+		if err != nil {
+			writeError(w, fmt.Errorf("unable to parse float64 from: %s, err:", err))
```
Comment (trekhopton) [inline cmd/oceanbench/set.go:422]: Message needs to be restructure as above.

**Review by trekhopton — CHANGES_REQUESTED** [review]

File: cmd/oceanbench/set.go [hunk 3]
```diff
@@ -374,6 +376,99 @@ func editDevicesHandler(w http.ResponseWriter, r *http.Request) {
 	http.Redirect(w, r, "/set/devices?ma="+ma, http.StatusFound)
 }
 
+// configDevicesHandler handles configuration of new devices.
+//
+// Form Fields:
+//
+//	name = Name of the new device
+//	ma = MAC address
```
Comment (scruzin) [inline cmd/oceanbench/set.go:384]: For consistency, please use standard two-letter query params, i.e., `ma` for MAC address, `ct` for client type, `wi` for the command separated string ssid-name, ssid-password, `ll` for comma-separated latitude,longitude, etc.

**Review by scruzin — CHANGES_REQUESTED** [review]

File: cmd/oceanbench/set.go (hunk 3 above)
Comment (scruzin) [reply cmd/oceanbench/set.go:384]: PS These query params form part of our REST API, that is why it's important to keep them consistent.

Comment (ao-david) [reply cmd/oceanbench/set.go:384]: done

File: cmd/oceanbench/set.go (hunk 1 above)
Comment (ao-david) [reply cmd/oceanbench/set.go]: done

**Review by trekhopton — APPROVED** [review]

**Pull request merged**